src/message/handleMessage.ts

```diff
@@ -0,0 +1,56 @@
+import { Platform } from 'react-native'
+import { Dispatch } from 'redux'
+
+import { AnalyticsMessage } from '../types/analytics'
+
+import { messageHandlers as analytics } from './handlers/analytics'
+import { messageHandlers as android } from './handlers/android'
+import { messageHandlers as audio } from './handlers/audio'
+import { messageHandlers as cast } from './handlers/cast'
+import { messageHandlers as haptics } from './handlers/haptics'
+import { messageHandlers as lifecycle } from './handlers/lifecycle'
+import { messageHandlers as linking } from './handlers/linking'
+import { messageHandlers as logging } from './handlers/logging'
+import { messageHandlers as notification } from './handlers/notification'
+import { messageHandlers as oauth } from './handlers/oauth'
+import { messageHandlers as search } from './handlers/search'
+import { messageHandlers as theme } from './handlers/theme'
+import { messageHandlers as version } from './handlers/version'
+import { Message, MessageHandlers } from './types'
+
+const isIos = Platform.OS === 'ios'
+
+const messageHandlers: Partial<MessageHandlers> = {
+  ...analytics,
+  ...(isIos ? {} : android),
+  ...audio,
+  ...cast,
+  ...haptics,
+  ...lifecycle,
+  ...linking,
+  ...logging,
+  ...notification,
+  ...oauth,
+  ...search,
+  ...theme,
+  ...version
+}
+
+export const handleMessage = async (
+  message: Message | AnalyticsMessage,
+  dispatch: Dispatch,
+  postMessage: (message: Message) => void,
+  reload: () => void
+) => {
+  const handler = messageHandlers[message.type]
+  if (handler) {
+    handler(message, dispatch, postMessage, reload)
```

Comment:
I think this would be a bit nicer if we did 

{ message, dispatch, postMessage, reload }

instead of position args. would remove a lot of the `_` right?

Also, might be worth adding a quick comment here on what these things do (e.g. that `postMessage` is your way to respond back)

Reply:
Agreed, I'm a big fan of the `options` argument pattern on anything over arity 3. Might add that to the js styleguide at some point

Reply:
i think even at 1 thing the options arg is better. so much more extensible